### PR TITLE
DT-1093 NomsId Algorithm - Logic only no persistence

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+                return NomsIdSequence.builder()
+                    .suffixAlphaSeq(27)
+                    .currentSuffix(lvSuffixValue)
+                    .currentPrefix(lvPrefixValue)
+                    .prefixAlphaSeq(lvPrefixSeqValue)
+                    .nomsId(0)
+                    .build();
+
+            } else {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java:157]: isn't this `else` unnecessary as previous line has a `return`?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+                return NomsIdSequence.builder()
+                        .suffixAlphaSeq(lvSuffixSeqValue)
+                        .currentSuffix(lvSuffixValue)
+                        .currentPrefix(currentPrefix)
+                        .prefixAlphaSeq(prefixAlphaSeq)
+                        .nomsId(0)
+                        .build();
+            }
+        } else {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java:179]: also with this `else`

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    private final String currentSuffix;
+
+    public NomsIdSequence next() {
+
+        var lvPrefixValue = alphabetic(prefixAlphaSeq);
+        var lvSuffixValue = alphabetic(suffixAlphaSeq);
+        final var lvReturnValue = lvPrefixValue + StringUtils.leftPad(String.valueOf(nomsId + 1), 4, "0") + lvSuffixValue;
+
+        if (lvReturnValue.equals(currentPrefix + "9999" + currentSuffix)) {
+            if (suffixAlphaSeq == 702) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java:140]: what's the magic `702`?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Builder
+@Data
+@EqualsAndHashCode(callSuper = false)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java:16]: don't need `callSuper` as no super class?

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+@AllArgsConstructor
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java:13]: do you need the all args if you've specified a builder?

Comment (Mjwillis) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/NomsIdSequence.java]: Resolved

**Comment by Mjwillis** [comment]
@petergphillips  updated

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**